### ISSUE TITLE
Completion woes

### ISSUE THREAD

**Issue by thofma** [thofma/Hecke.jl#1075]
```julia
julia> Qx, x = QQ["x"];

julia> f = x^9 - 828*x^7 - 4371*x^6 + 226071*x^5 + 2371023*x^4 - 14243253*x^3 - 318645900*x^2 - 1637156439*x - 2754662093;

julia> K, a = number_field(f, "a");

julia> OK = maximal_order(K);

julia> p = 3;

julia> P = prime_ideals_over(OK, p)[1];

julia> C, mC = completion(K, P, 8);

julia> b = 5001310657//440423231859045*a^8 - 332069942701//3963809086731405*a^7 - 34477045500619//3963809086731405*a^6 + 22827170414018//1321269695577135*a^5 + 1893757018539416//792761817346281*a^4 + 29698097663762398//3963809086731405*a^3 - 57718358174700707//264253939115427*a^2 - 1362121503883347224//792761817346281*a - 13294109890580232283//3963809086731405;

julia> mC(b)
0

julia> valuation(b, P)
0
```

This does not look good. What is our guarantee for `mC(b)`?

**Comment by fieker** [comment]
I'll have a look tomorrow. I think this is already fixed locally....

**Comment by fieker** [comment]
On Thu, May 18, 2023 at 08:00:47AM -0700, Tommy Hofmann wrote:
> ```julia
> julia> Qx, x = QQ["x"];
> 
> julia> f = x^9 - 828*x^7 - 4371*x^6 + 226071*x^5 + 2371023*x^4 - 14243253*x^3 - 318645900*x^2 - 1637156439*x - 2754662093;
> 
> julia> K, a = number_field(f, "a");
> 
> julia> OK = maximal_order(K);
> 
> julia> p = 3;
> 
> julia> P = prime_ideals_over(OK, p)[1];
> 
> julia> C, mC = completion(K, P, 8);
> 
> julia> b = 5001310657//440423231859045*a^8 - 332069942701//3963809086731405*a^7 - 34477045500619//3963809086731405*a^6 + 22827170414018//1321269695577135*a^5 + 1893757018539416//792761817346281*a^4 + 29698097663762398//3963809086731405*a^3 - 57718358174700707//264253939115427*a^2 - 1362121503883347224//792761817346281*a - 13294109890580232283//3963809086731405;
> 
> julia> mC(b)
> 0
> 
> julia> valuation(b, P)
> 0
> ```
> 
> This does not look good. What is our guarantee for `mC(b)`?

I don't know - what should it be? The problem here is that the index is
divisible by 3^21. Since the maps are defined on the field level, I'd
guess that in order to safely have precision 8, the map needs to use 29
or more.

But more to the point: what do we guaratee? b as a field element has
coefficients that are 0 in the precision (8 -> 2 or 3 in the p/q-adic
field)

If we complete rings, then the sematic might be easier.

Or we could silently add the index

**Comment by fieker** [comment]
On Thu, May 18, 2023 at 08:00:47AM -0700, Tommy Hofmann wrote:
> ```julia
> julia> Qx, x = QQ["x"];
> 
> julia> f = x^9 - 828*x^7 - 4371*x^6 + 226071*x^5 + 2371023*x^4 - 14243253*x^3 - 318645900*x^2 - 1637156439*x - 2754662093;
> 
> julia> K, a = number_field(f, "a");
> 
> julia> OK = maximal_order(K);
> 
> julia> p = 3;
> 
> julia> P = prime_ideals_over(OK, p)[1];
> 
> julia> C, mC = completion(K, P, 8);
> 
> julia> b = 5001310657//440423231859045*a^8 - 332069942701//3963809086731405*a^7 - 34477045500619//3963809086731405*a^6 + 22827170414018//1321269695577135*a^5 + 1893757018539416//792761817346281*a^4 + 29698097663762398//3963809086731405*a^3 - 57718358174700707//264253939115427*a^2 - 1362121503883347224//792761817346281*a - 13294109890580232283//3963809086731405;
> 
> julia> mC(b)
> 0
> 
> julia> valuation(b, P)
> 0
> ```
> 
> This does not look good. What is our guarantee for `mC(b)`?
> 

P.S.:

julia> C, mC = completion(K, P, 8+21);

julia> mC(b)
((2*3^0 + 2*3^1 + 3^2 + 3^3 + O(3^4))*a^2 + (3^0 + 3^2 + O(3^4))*a + 3^0
+ 3^3 + O(3^4))*a^2 + ((2*3^0 + 2*3^1 + 2*3^3 + O(3^4))*a^2 + (3^0 + 3^1
+ 3^2 + 3^3 + O(3^4))*a + 3^2 + 3^3 + O(3^4))*a + (2*3^1 + 3^2 + 3^3 +
O(3^4))*a^2 + (2*3^0 + 3^2 + O(3^4))*a + 2*3^0 + 3^1 + 2*3^2 + O(3^4)

**Comment by thofma** [comment]
Fix in #1077
